Data/ByteString/Internal.hs

```diff
@@ -245,14 +253,34 @@ unsafePackLenChars len cs0 =
 --
 unsafePackAddress :: Addr# -> IO ByteString
 unsafePackAddress addr# = do
+#if __GLASGOW_HASKELL__ >= 811
+    return (PS (ForeignPtr addr# FinalPtr) 0 (I# (cstringLength# addr#)))
+#else
     p <- newForeignPtr_ (castPtr cstr)
     l <- c_strlen cstr
-    return $ PS p 0 (fromIntegral l)
+    let len = fromIntegral l
+    return $ PS p 0 len
```

Comment:
This cosmetic change is unnecessary, and will conflict with at least one pending PR, perhaps best to leave it out.